command/flag_kv.go

```diff
@@ -129,6 +129,14 @@ func (v *FlagStringSlice) Set(raw string) error {
 	return nil
 }
 
+var (
+	// This regular expression is how we check if a value for a variable
+	// matches what we'd expect a rich HCL value to be. For example: {
+	// definitely signals a map. If a value DOESN'T match this, we return
+	// it as a raw string.
+	varFlagHCLRe = regexp.MustCompile(`^["\[\{]`)
```

Comment:
I think previously we would have allowed whitespace at the start of the string, so this is somewhat different.